### ISSUE TITLE
Bring Back Map Rotation

### ISSUE THREAD

**Issue by ghost** [d3athrow/vgstation13#7353]
AAAAAAAAAAAAAA

**Comment by Imagebard** [comment]
[Not a bug]
N3X15 removed map rotation. Its random now.

**Comment by ghost** [comment]
> not a bug

**Comment by SysadminJeroen** [comment]
Can we or can we not vote for the next map?

**Comment by Imagebard** [comment]
**No, its random now.**

**Comment by SysadminJeroen** [comment]
This can be a good thing, other maps except Box will be updated which could make the maps more fun.

**Comment by Killette2** [comment]
Except taxi isn't fun. I don't want to have to tread through shit to make it less shitty, when we already have 3 other maps that are already good.

**Comment by SysadminJeroen** [comment]
@Killette2 Opinions.
I could find Taxi more enjoying than Deficiency or Meta.

**Comment by ihadtoregisterforthis** [comment]
At least give it a trial period

**Comment by Killette2** [comment]
There's nothing to do on taxi. You can't build shit, often your department is lacking things, that even if you made a git about it there's more than often no room for it, taxi offers nothing special. It's sole purpose is to be a bad map.

**Comment by PJB3005** [comment]
It's a gimmick, a poorly executed one at that.

I agree it's got its issues.

**Comment by Killette2** [comment]
I think we should just scrap the whole map and try the taxi gimmick with a better map. The taxi's aren't the issues, you could put taxi's on any improved map and it would more than likely be better received.

**Comment by SysadminJeroen** [comment]
@Killette2 As I have commented before, this gives maps that usually aren't getting played more of a chance. If a map is getting played more then it is more likely to get improved.
@PJB3005 The gimmick of Taxi that I like is that you need to ferry over to other stations if you want to reach certain departments.

**Comment by Killette2** [comment]
@Jeroen52 If the gimmick was just ferrying over to other stations then taxi wouldn't be a bad map. The bad part is that everything is fucking small, and not only that but poorly connected (see kitchen/bar/theater), shit is missing, and there's no real maintenance tunnels.

**Comment by SysadminJeroen** [comment]
@Killette2 I don't think that Taxi Station was designed for more than 32 players, I have seen players starting to get cramped at the taxi's.

If you see anything missing you can edit the map and add it yourself, or add it to the issues list in this repo and wait for someone to do it.

The kitchen, bar and theater are a mess, someone should sort it out.

And some players are getting around the part that there are no maintenance tunnels by going EVA.

If someone is planning to make a station that is like Taxi but with fixes, I would certainly like to see it.

**Comment by Uqualm** [comment]
I'm glad that I no longer play on /vg/station, because this is absolutely unacceptable and would cause me to immediately leave in disgust if I were still playing. When I think of coders that make drastic changes against the wishes of the majority of the player base, my mind focuses on 2012-2013 era /tg/station. For all the gripes one can have with /vg/ station, the coders themselves aren't usually the issue. Now things are changing, a new precedent has been set. The more experienced coders will exercise their authority more often and in more substantial ways, often in conflict with what the players want. 

Let us look at what it is Nexis has actually done has done with this change. At the end of every round, the players are asked to vote for which map they would like to play on next. Boxstation is chosen almost every single time, and beats out the other stations by a large margin. From this we know very clearly what it is the majority want.

**Comment by Uqualm** [comment]
I'm glad that I no longer play on /vg/station, because this is absolutely unacceptable and would cause me to immediately leave in disgust if I were still playing. When I think of coders that make drastic changes against the wishes of the majority of the player base, my mind focuses on 2012-2013 era /tg/station. For all the gripes one can have with /vg/ station, the coders themselves aren't usually the issue. Now things are changing, a new precedent has been set. The more experienced coders will exercise their authority more often and in more substantial ways, often in conflict with what the players want.

Let us look at what it is Nexis has actually done has done with this change. At the end of every round, the players are asked to vote for which map they would like to play on next. Box Station is chosen almost every single time. From this we know very clearly what it is the majority want. An overwhelming majority of players prefer Box Station. If you look at the /vg/ thread, you'll see a different story. People make posts making fun of Box players, and calling them "boxbabies" and begging to be able to play other maps. From the data we have we can only conclude that these posts, numerous as they may seem to be, are nothing more than posts by a very vocal minority.

So Nexis believes that this is a problem, people should want to play on other maps, and will if we just "fix" them. According to Nexis, the only way to "fix" these maps is to find out why people don't want to play them, and the only way to do that is to make them play them. Presumably instead of just leaving, or coming back when Box is the map, they will magically report their grievances in the form of a bug report. After this is done for a while, the map will magically be "fixed" and all will be well and everyone won't want to play on Box. Or perhaps I'm wrong, and the intent isn't to incentivize people to report bugs, but more of a threat: "You won't be able to play on Box again until you report some bugs. Get to it." If that is the case, the situation is much worse than I thought. Continuing down this line of thought, could this instead mean that Nexis himself wanted to play on other maps and willed it so?  He figured it was within his means to force other people to play them to satiate his desires and so he did it, without a care for anyone else.

The results of this are clear regardless of the reasoning Nexis used. The vast majority of people have had their opinions tossed aside in favor of a privileged opinion of a singular person. If you consider that this was in the realm of what what was traditionally a democratic process, the word "tyranny" comes to mind. I can only imagine that Nexis has reacted in smug condescension to any objections raised to this, refusing to seriously argue for his case. 

One last thing, I will discuss is the logic used by Nexis to justify the change. He says that the only way to fix the maps is to play them so flaws are identified, and so we must force players to play on them. This isn't very convincing for a few reasons:

1. It assumes a priori, that the maps themselves are able to be fixed, and are not at their core, flawed. In that, if you were to make the changes needed to fix say, Taxi Station, it would cease being Taxi Station altogether and simply be a poor emulation of Box. 

2. It assumes that the only way, or the best way, to find and fix flaws with the other maps is to force people to play them.

3. There is no guarantee that people will report bugs instead of just leaving to do something else when they are forced to play on another map.

I could say more really, but I thought I would keep it succinct I think that covers the bare minimum of what you need to know to understand the situation.

**Comment by N3X15** [comment]
tl;dr

**Comment by Senaat1** [comment]
While i think this is a bad idea

fuck off sarca

**Comment by 9600bauds** [comment]
fuck off sarca

**Comment by SysadminJeroen** [comment]
fuck off sarca

**Comment by optimumtact** [comment]
fuck off sarca 2016

**Comment by ihadtoregisterforthis** [comment]
Free replies here

**Comment by 9600bauds** [comment]
making bug reports is not "magic"
PRs that address those bug reports are not "magic"
i'm not sure what would lead you to think that either of these don't already happen

**Comment by ghost** [comment]
fuck off sarca

**Comment by optimumtact** [comment]
stick with map rotation random, we have it on tg and after a few weeks bitching it was okay, our was weighted by user pref choice though, so box does see higher play counts.

**Comment by Intigracy** [comment]
> It assumes that the only way, or the best way, to find and fix flaws with the other maps is to force people to play them.

But this is true?

**Comment by 9600bauds** [comment]
Well on the plus side, we now know that mappers are literally fucking magic.

**Comment by SysadminJeroen** [comment]
@oranges
That could help with the bitching.
More box, but also the other maps.
And NO TAXI when there are too many players.

2015-12-31 0:05 GMT+01:00 oranges notifications@github.com:

**Comment by SysadminJeroen** [comment]
@Intigracy
I have already submitted a possible bug on Taxi which I would only submit
as we play there more often.

2015-12-31 0:06 GMT+01:00 Intigracy notifications@github.com:

**Comment by Intigracy** [comment]
Yeah I know I was responding to sarca's point implying that this isn't true.

**Comment by optimumtact** [comment]
yeah we added some player count limits, although I think it was mostly for birdboat and ministation

**Comment by Zth--** [comment]
Posting in ebin thread........... (not, fuck off sarca)

**Comment by clusterfack** [comment]
Fuck off sarca

**Comment by ghost** [comment]
You can make fun of sarca and meme all you want, he's right though. Everyone including collaborators should go through peer review on game additions and substantial changes such as these.

What pisses me off most is that @N3X15 straight up disregarded any form of quality control and just merged his code without any form of feedback from the community, yet he happily slaps [controversy] on pull requests and lets them rot for weeks. It's not even the first time he did it, see coal golems and derelict MoMMI names.

It's hypocritical and something a real asshole would do.

**Comment by Senaat1** [comment]
fuck off termed

**Comment by Intigracy** [comment]
Sorry but it's actually working, we're actually getting bug reports for the maps now that they're being played.

ty eggman of coding

The system could use some adjustments probably, like playercaps for the maps or weighted selection, but for now it's fine while the kinks get ironed out.

tl;dr
fuck off sarca

**Comment by Senaat1** [comment]
In all seriousness though, i agree with you on that, there should be transparancy and discussion on this kind of stuff ESPECIALLY as it's as controversial as it is instead of him just merging it.

**Comment by ghost** [comment]
The maps are not unpopular because of being bugged, they are unpopular by the virtue of being designed with a completely different mindset compared to box.

Deff is a maint gimmick, taxi is the most enraging gimmick, meta is the most similar to box with switched departments and is just way too big.

I would propose a vote system where after each round a map is assigned a % chance based on the proportion of votes it got in that particular vote. This way even the minority maps would get played proportionately to their playerbase.

**Comment by 9600bauds** [comment]
meta is literally smaller than box

**Comment by PJB3005** [comment]
Pretty sure meta is smaller, but it feels more roomy.

Also, can we fucking disallow coders from directly pushing shit unless it's either fixes or OH GOD things

**Comment by clusterfack** [comment]
Fuck off termed

**Comment by Cisrox** [comment]
this is bullshit, map vote when
